src/core/ext/filters/client_channel/subchannel.cc

```diff
@@ -775,11 +781,20 @@ static void recv_trailing_metadata_ready(void* arg, grpc_error* error) {
   get_call_status(call, md_batch, GRPC_ERROR_REF(error), &status);
   grpc_core::channelz::SubchannelNode* channelz_subchannel =
       call->connection->channelz_subchannel();
-  GPR_ASSERT(channelz_subchannel != nullptr);
-  if (status == GRPC_STATUS_OK) {
-    channelz_subchannel->RecordCallSucceeded();
-  } else {
-    channelz_subchannel->RecordCallFailed();
+  if (channelz_subchannel != nullptr) {
+    if (status == GRPC_STATUS_OK) {
+      channelz_subchannel->RecordCallSucceeded();
+    } else {
+      channelz_subchannel->RecordCallFailed();
+    }
+  }
+  if (*call->lb_recv_trailing_metadata_ready != nullptr) {
```

Comment:
As currently structured, if retries are enabled, we will actually run this twice: once here, and then again in the client_channel code.

I don't think this functionality should be in the subchannel code at all, since the subchannel does not interact with the LB policy; all of the LB policy interaction is handled in the client_channel code.